service/src/main/java/uk/nhs/adaptors/gp2gp/common/storageconnectors/AzureStorageConnector.java

```diff
@@ -0,0 +1,39 @@
+package uk.nhs.adaptors.gp2gp.common.storageconnectors;
+
+import java.io.ByteArrayOutputStream;
+import java.io.IOException;
+import java.io.InputStream;
+import java.io.OutputStream;
+import org.springframework.beans.factory.annotation.Autowired;
+import com.azure.storage.blob.BlobClient;
+import com.azure.storage.blob.BlobContainerClient;
+
+public class AzureStorageConnector implements StorageConnector {
+
+    @Autowired
+    private BlobContainerClient containerClient;
```

Comment:
Are you sure this works? This property is autowired but the StorageConnectorFactor instantiates it with `new`. I don't see how this field would be injected.

Reply:
I have just tested this and it still appears to be working correctly.

Reply:
Nevermind, this is some spring magic. Dependency injection happens after FactoryBean returns its instance apparently